### PR TITLE
Update escrow to feature metadata, recipient can be set after creation

### PR DESCRIPTION
**Pull request by Callum-A**
After discussions with @JakeHartnell and @orkunkl we decided to expand functionality for escrow, this new functionality is as follows:

- Recipient is now optional upon creation, this allows Escrows to be used as bounties e.g. for bug bounties etc.
- Recipient can be set by the arbiter using the new SetRecipient message
- Escrows now feature additional meta data, including a title and description. E.g. for a bounty the description may include steps needed to be valid for said bounty.

Tests are added/updated accordingly.

Feedback as always is welcome

### CODE REVIEW

File: contracts/cw20-escrow/src/contract.rs
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:146]:
```suggestion
        ("id", id),
```

File: contracts/cw20-escrow/src/contract.rs
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:147]:
```suggestion
        ("recipient", recipient),
```

**Comment by Callum-A** [comment]
Changes all look good to me : )

**Comment by orkunkl** [comment]
Waiting for Circle CI to be configured by core team.

File: contracts/cw20-escrow/src/contract.rs
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:480]:
```suggestion


```

**Comment by orkunkl** [comment]
Circle CI works now, can you fix the errors @Callum-A and merge?

**Review by orkunkl — APPROVED** [review]

**Comment by Callum-A** [comment]
Will do

**Comment by Callum-A** [comment]
@orkunkl any update on this? Build is fixed for escrow however other contracts are breaking it.

**Pull request merged**
